app/controllers/settings_controller.rb

```diff
@@ -4,9 +4,7 @@ class SettingsController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_view_your_settings")
   end
 
-  before_filter :except => :unsubscribe do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_view_this_content")
-  end
+  ensure_can_access_person :person_id, except: [:unsubscribe]
```

Comment:
Hmmm... I guess you mean it's used in only one other place in *this controller*?

I too think that this parameter name should be changed. However, it's not a low hanging fruit. In `routes.rb` the `people` resource contains a lot of nested resources inside of it:

```
resources :people do
  resources :messages
  resources :settings
  resources :listings
  resources :testimonials
  ...
end
```

So if we change the param name from `person_id` to `person_username` by changing the people resource to:

```
resources :people, param: :username do
```

...then it will change the param name for *all the nested resources*, meaning that we would need to go through all the controllers accessing this parameter name, and all the code that creates URLs pointing to these resources (e.g. `person_settings_path(person_id: ... )` needs to be changed to `person_settings_path(person_username: ... )`).

With a quick search I found 60+ places where renaming is needed.